mono/metadata/icall.c

```diff
@@ -1846,7 +1846,11 @@ ves_icall_MonoField_SetValueInternal (MonoReflectionField *field, MonoObject *ob
 				 * This is complicated by the fact that Nullables have
 				 * a variable structure.
 				 */
-				nullable = mono_object_new (mono_domain_get (), nklass);
+				nullable = mono_object_new_checked (mono_domain_get (), nklass, &error);
+				if (!mono_error_ok (&error)) {
+					mono_error_set_pending_exception (&error);
```

Comment:
What if mono_error_set_pending_exception returned a boolean so we could write this change like this:
```
if (mono_error_set_pending_exception (&error))
   return
```

CC @ludovic-henry